hide/client/hide_client.py

```diff
@@ -60,7 +61,9 @@ def run_task(
             payload["alias"] = alias
 
         response = requests.post(
-            f"{self.base_url}/projects/{project_id}/tasks", json=payload
+            f"{self.base_url}/projects/{project_id}/tasks",
+            json=payload,
+            headers={"X-Timeout-Seconds": str(timeout_seconds)},
```

Comment:
In general I think we should use `timeout` arg in post(). On the server side it will be as simple as handling context cancellations, which we already do. On time out client should disconnect and context cancelled. However, with docker exec. This will not work with current solution.